test/Microsoft.AspNetCore.HttpsEnforcement.Tests/HstsMiddlewareTests.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+namespace Microsoft.AspNetCore.HttpsPolicy.Tests
+{
+    public class HstsMiddlewareTests
+    {
+        [Fact]
+        public async Task SetOptions_DefaultsSetCorrectly()
+        {
+            var builder = new WebHostBuilder()
+                .UseUrls("https://*:5050")
+                .ConfigureServices(services =>
+                {
+                })
+                .Configure(app =>
+                {
+                    app.UseHsts();
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello world");
+                    });
+                });
+
+            var server = new TestServer(builder);
+            var client = server.CreateClient();
+            client.BaseAddress = new Uri("https://localhost:5050");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            // By default, we will not redirect if we only use the Hsts Middleware.
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal("max-age=0", response.Headers.GetValues(HeaderNames.StrictTransportSecurity).FirstOrDefault());
+        }
+
+        [Theory]
+        [InlineData(0, false, false, "max-age=0")]
+        [InlineData(-1, false, false, "max-age=-1")]
+        [InlineData(0, true, false, "max-age=0; includeSubDomains")]
+        [InlineData(50000, false, true, "max-age=50000; preload")]
+        [InlineData(0, true, true, "max-age=0; includeSubDomains; preload")]
+        [InlineData(50000, true, true, "max-age=50000; includeSubDomains; preload")]
+        public async Task SetOptions_ConfigureOptions(int maxAge, bool includeSubDomains, bool preload, string expected)
+        {
+            var builder = new WebHostBuilder()
+                .UseUrls("https://*:5050")
+                .ConfigureServices(services =>
+                {
+                    services.Configure<HstsOptions>(options => {
+                        options.Preload = preload;
+                        options.IncludeSubDomains = includeSubDomains;
+                        options.MaxAge = maxAge;
+                    });
+                })
+                .Configure(app =>
+                {
+                    app.UseHsts();
+                    app.Run(context =>
+                    {
+                        return context.Response.WriteAsync("Hello world");
+                    });
+                });
+
+            var server = new TestServer(builder);
+            var client = server.CreateClient();
+            client.BaseAddress = new Uri("https://localhost:5050");
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            // By default, we will not redirect if we only use the Hsts Middleware.
```

Comment:
You copy pasted this comment below and doesn't make sense there. I don't think this adds a lot of value here, so I would just remove it.